packages/core/core/src/Strapi.ts

```diff
@@ -269,20 +270,19 @@ class Strapi extends Container implements Core.Strapi {
       .add('entityValidator', entityValidator)
       .add('entityService', () => createEntityService({ strapi: this, db: this.db }))
       .add('documents', () => createDocumentService(this))
-      .add(
-        'db',
-        () =>
-          new Database(
-            _.merge(this.config.get('database'), {
-              logger,
-              settings: {
-                migrations: {
-                  dir: path.join(this.dirs.app.root, 'database/migrations'),
-                },
+      .add('db', () => {
+        const outDir = tsUtils.resolveOutDirSync(this.dirs.app.root) ?? this.dirs.app.root;
+        return new Database(
+          _.merge(this.config.get('database'), {
+            logger,
+            settings: {
+              migrations: {
+                dir: path.join(root, 'database/migrations'),
```

Comment:
Did you mean `outDir` instead? :thinking:

Reply:
This is breaking the build

Reply:
Fixed.